logger.go

```diff
@@ -0,0 +1,175 @@
+package glog
+
+import (
+	"context"
+	"fmt"
+)
+
+// Logger provides logging functionality with additional data and prefixing.
+type Logger struct {
+	*loggingT
+	prefix string
+	// Extra arguments to be appended to each request
+	data []interface{}
+}
+
+// NewLogger creates a Logger instance with no additional data.
+func NewLogger() Logger {
+	return Logger{
+		loggingT: &logging,
+	}
+}
+
+// WithContext creates a logger from a context.Context
+func WithContext(ctx context.Context) Logger {
+	data, _ := ctx.Value(contextKeyData).([]interface{})
+	prefix, _ := ctx.Value(contextKeyPrefix).(string)
+	return Logger{
+		loggingT: &logging,
+		data:     data,
+		prefix:   prefix,
+	}
+}
+
+// WithPrefix creates a Logger with a given prefix.
+func WithPrefix(prefix string) Logger {
+	return Logger{
+		loggingT: &logging,
+		prefix:   prefix,
+	}
+}
+
+// WithData creates a Logger with a given set of backend data.
```

Comment:
Unrelated to this particular location: in glog_backend.go, in the Data(arg interface) function, I would add a check that the provided argument is not data already. That is:
if _, ok := arg.(data); ok {
  return arg
}
return data{d:arg}

Reply:
Interesting thought, but I'm not sure of what the implications of that might be if someone is already misusing it somewhere. Probably out of scope for this change.

Reply:
Well, right now it would package the data like so:
for glog.Data(glog.Data(arg)) it would look like:
d: {
  d: {
    d: arg
  }
}
My suggestion will make sure it's only one level deep. That will come handy as people will be using the new logger and may forget they don't need to do glog.Data and will still do that.